src/foam/dao/pg/PostgresDAO.java

```diff
@@ -0,0 +1,176 @@
+package foam.dao.pg;
+
+import foam.core.*;
+import foam.dao.DAO;
+import foam.dao.ProxyDAO;
+
+import java.sql.*;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+
+public class PostgresDAO extends ProxyDAO {
```

Comment:
It shouldn't extend ProxyDAO, it should extend AbstractDAO. This is a real DAO implementation (Strategy), not a decorator or proxy.